src/python/gem5/resources/workload.py

```diff
@@ -24,219 +24,55 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-from .resource import obtain_resource
+from .resource import obtain_resource, WorkloadResource
 from .client import get_resource_json_obj
 
 from _m5 import core
+from m5.util import warn
 
 from typing import Dict, Any, List, Optional
 
 
-class AbstractWorkload:
+def CustomWorkload(function: str, parameters: Dict[str, Any]):
     """
-    Workloads contain information needed to build a workload.
+    A custom workload gem5 resource. It can be used to specify a custom,
+    local workload.
 
-    A workload specifies a function and its parameters to run on a board to
-    set a workload. Workload's are passed to board via the `AbstractBoard`'s
-    `set_workload` function.
-
-    The `AbstractBoard` has a `set_workload` function which accepts an
-    AbstractWorkload. The `set_workload` function uses the `get_function_str`
-    to determine which function should be called on the board and the
-    `get_parameters` function specifies the parameters to be passed.
-
-    Example
-    -------
-
-    ```py
-    workload = CustomWorkload(
-        function = "set_se_binary_workload",
-        parameters = {
-            "binary" : obtain_resource("x86-print-this"),
-            "arguments" : ["hello", 6]
-        },
+    **Warning**: This class is deprecated and changed to a funtion and will be removed in future
+    releases of gem5. Please use the `WorkloadResource` class instead. This
+    class is merely a wrapper for it.
+    """
+    warn(
+        "The `CustomWorkload` class is deprecated. Please use "
+        "`WorkloadResource` instead."
     )
+    return WorkloadResource(function=function, parameters=parameters)
 
-    board.set_workload(workload)
-    ```
-
-    The above is the equivalent of:
 
-    ```py
-    board.set_se_binary_workload(
-        binary = obtain_resource("x86-print-this"),
-        arguments = ["hello", 6],
-    )
-    ```
-
-    Notes
-    -----
-    This class should not be used directly. Please use `Workload` or
-    `CustomWorkload`.
+def Workload(
+    workload_name: str,
+    resource_directory: Optional[str] = None,
+    resource_version: Optional[str] = None,
+    clients: Optional[List] = None,
+    gem5_version: Optional[str] = core.gem5Version,
+):
     """
+    This function was created to maintain backwards compability for v23.0.0
+    and prior releases of gem5 where `Workload` was a class.
```

Comment:
```suggestion
    **Warning**: The `Workload` class is deprecated. It will be removed in a
    future release of gem5. Please use the `gem5.resources.resource.WorkloadResource`
    class instead.
```